hws/src/dr_hw_resources.py

```diff
@@ -136,104 +138,95 @@ def load_to_db(self):
 
 class dr_parse_argument():
     def __init__(self, data):
-        keys = ["mlx5dr_debug_res_type", "index"]
+        keys = ["mlx5dr_debug_res_type", "arg_index", "index", "data"]
         self.data = dict(zip(keys, data + [None] * (len(keys) - len(data))))
-        self.args_arr = data[2:] if (len(data) > 2) else []
 
     def get_index(self):
         return self.data.get("index")
 
     def load_to_db(self):
-        _db._argument_db[self.get_index()] = self.args_arr
+        _db._argument_db[self.get_index()] = self.data.get("data")
 
 
 def dr_parse_fw_modify_argument_set(raw):
-    return "data: %s" % hex(int(raw[8:16], 16))
+    raw = raw[8:16]
+    text = "data: %s" % hex(int(raw, 16))
+    return {"text": text, "raw": raw}
 
 
-def dr_parse_fw_modify_pattern_set(raw):
-    field = modify_pattern_field_dic.get(int(raw[1:4], 16))
-    offset = hex(int(raw[4:6], 16) & 0x1f)
-    length = int(raw[6:8], 16) & 0x1f
-    length = "0x20" if length == 0 else hex(length)
-    text = "SET: field: %s offset: %s length: %s" %\
-            (field, offset, length)
+def dr_parse_modify_pattern_set(raw):
+    _raw = hex_to_bin_str(raw, 64)
+    action = dr_parse_set_action(_raw[0:32], _raw[32:64], False)
 
-    return {"type": 0x1, "text": text}
+    return {"type": 0x1, "text": action_pretiffy(action, False), "raw": raw[0:8]}
```

Comment:
I don't understand... You removed action_pretiffy function from this file, do you import it somewhere?

Reply:
yes, i import at the begining of the file:
from src.dr_common_functions import *